src/datasets/domain/models/DatasetPreviewSubset.ts

```diff
@@ -0,0 +1,6 @@
+import { DatasetPreview } from './DatasetPreview';
+
+export interface DatasetPreviewSubset {
+  datasetPreviews: DatasetPreview[];
+  totalDatasetCount: number;
+}
```

Comment:
I believe that for the sake of consistency within the module, `getDatasetFiles` should also return FilesSubset, including files and the count information

This would ensure that the module's responses are predictable. Users of the module would then know that if they use some `getAllItems` use case, the response will be paginated and will include both the items and their count

I think that the module use cases shouldn't be coupled to the type of API used, Native API or Search API, and the module should be consistent even if each API works differently

Reply:
I agree, but I wouldn't make this change now as I would first extend the files API to return the total files count along with the files to avoid making two API calls per use case call. We can create two new issues for this: Files API extension + files use case extension.

Reply:
https://github.com/IQSS/dataverse/issues/10202
https://github.com/IQSS/dataverse-client-javascript/issues/109